nowcasting_dataset/data_sources/nwp/nwp_data_source.py

```diff
@@ -128,6 +128,11 @@ def _post_process_example(self, selected_data: xr.Dataset, t0_dt: pd.Timestamp)
         start_dt = self._get_start_dt(t0_dt)
         end_dt = self._get_end_dt(t0_dt)
 
+        # if t0_dt is not on the hour, e.g. 13.05.
+        # Then if the history_minutes is 1 hours,
+        # so start_dt will be 12.05, but we want to the 12.00 time step too
+        start_dt = start_dt.floor("H")
```

Comment:
Very good catch; and very elegant fix!  Nice!

(And perfect timing too, because I think I just bumped into the same bug: #410 :slightly_smiling_face: )